client/main.lua

```diff
@@ -1,287 +1,563 @@
-            TaskReactAndFleePed(target, PlayerPedId())
-            local plate = QBCore.Functions.GetPlate(GetVehiclePedIsIn(target, true))
-            TriggerServerEvent('hud:server:GainStress', math.random(1, 4))
-            TriggerEvent('vehiclekeys:client:SetOwner', plate)
-            QBCore.Functions.Notify('You Got The Keys!', 'success')
-            Wait(10000)
-            IsRobbing = false
-        else
-            PoliceCall()
-            ClearPedTasks(target)
-            TaskReactAndFleePed(target, PlayerPedId())
-            TriggerServerEvent('hud:server:GainStress', math.random(1, 4))
-            QBCore.Functions.Notify('They Called The Cops!', 'error')
-            Wait(10000)
-            IsRobbing = false
+            isCarjacking = false
+            Wait(2000)
+            AttemptPoliceAlert("carjack")
+            Wait(Config.DelayBetweenCarjackings)
+            canCarjack = true
         end
+    end, function()
+        MakePedFlee(target)
+        isCarjacking = false
+        Wait(Config.DelayBetweenCarjackings)
+        canCarjack = true
     end)
 end
 
-local function IsBlacklistedWeapon()
-    local weapon = GetSelectedPedWeapon(PlayerPedId())
-    if weapon ~= nil then
-        for _, v in pairs(Config.NoRobWeapons) do
-            if weapon == GetHashKey(v) then
-                return true
+function AttemptPoliceAlert(type)
+    if not AlertSend then
+        local chance = Config.PoliceAlertChance
+        if GetClockHours() >= 1 and GetClockHours() <= 6 then
+            chance = Config.PoliceNightAlertChance
+        end
+        if math.random() <= chance then
+            local closestPed = GetNearbyPed()
+            if closestPed ~= nil then
+                local pedData = exports['cd_dispatch']:GetPlayerInfo()
+                pedData.type = type
+                pedData.sprite = 229
+                if type == "steal" then pedData.sprite = 380 end
+
+                --QBCore.Functions.Notify('Someone called the cops!', 'error')
+                AlertPolice(pedData)
             end
         end
+        AlertSend = true
+        SetTimeout(Config.AlertCooldown, function()
+            AlertSend = false
+        end)
     end
-    return false
 end
 
-local function DrawText3D(x, y, z, text)
+function AlertPolice(data) -- cd_dispatch
+    TriggerEvent('cd_dispatch:AddNotification', {
+        job_table = {'police', 'bcso', 'sasp'},
+        coords = data.coords,
+        title = '10-60 - Vehicle Theft',
+        message = "A "..data.sex.." was reported near "..data.street.." attempting to "..data.type.." a "..data.vehicle_colour.." "..data.vehicle_label.." ("..data.vehicle_plate..")",
+        flash = 0,
+        blip = {
+            sprite = data.sprite,
+            scale = 1.2,
+            colour = 3,
+            flashes = false,
+            title = '10-60 - Vehicle Theft',
+            time = (5*60*1000),
+            sound = 1,
+        }
+    })
+end
```

Comment:
Calling cd_dispatch, should be calling the default built in QBCore alerts (although not many really use it lul)